pkg/csource/csource_test.go

```diff
@@ -17,10 +16,6 @@ import (
 )
 
 func TestGenerate(t *testing.T) {
-	switch runtime.GOOS {
```

Comment:
```
--- FAIL: TestGenerate (11.23s)
    --- FAIL: TestGenerate/openbsd/amd64 (0.00s)
    	csource_test.go:51: seed=1541529791518546498
        --- FAIL: TestGenerate/openbsd/amd64/0/0 (0.41s)
        	csource_test.go:92: opts: {Threaded:false Collide:false Repeat:true RepeatTimes:0 Procs:0 Sandbox:none Fault:false FaultCall:0 FaultNth:0 EnableTun:false UseTmpDir:true EnableCgroups:false EnableNetdev:false ResetNet:false HandleSegv:false Repro:false Trace:false}
        		program:
        		mmap(&(0x7f0000000000/0x1000)=nil, 0x1000, 0x3, 0x1012, 0xffffffffffffffff, 0x0)
        		
        	csource_test.go:93: cpp failed: exec: "ecpp": executable file not found in $PATH
```

Which brings us to an interesting problem that we pretended does not exist so far.
All parameters in sys/targets/targets.go are not a property of the target OS only, they are a property of (build OS, target OS) pair. And there should be some notion of what target OSes are supported on a particular build OS.
Not sure if we will ever have ability to build target OS from more than 1 build OS (assuming that we drop *BSD build on linux). But we definitely can build more than 1 target OS on some build OSes (e.g. fuchsia/akaros intentionally only have cross-compilation from linux).

Perhaps we could start with adding `BuildOS string` to targets.osCommon. It should default to itself, but for fuchsia and akaros point to "linux". And then in this test we would check that runtime.GOOS == sysTarget.BuildOS. And probably something similar in Makefile to understand executors we need to build.

Reply:
I feel it will be unnecessarily burdensome to make it a requirement that everything builds on Linux. OpenBSD goes as far as intentionally not supporting cross-compiling any architectures beyond the initial bring up. We would be much better off maintaining a set of build bots if we want full presubmit coverage. Alternatively, like @mptre noted, we could rely on syzbot cluster to canary new commits and complain when things go awry.

Reply:
> Perhaps we could start with adding BuildOS string to targets.osCommon.

Sounds like a plan.

Reply:
Do you want to do this in the context of this PR?
Alternative we could do what Greg did in this PR (manually check host/target OS) and then do more systematic support in a separate PR.